spec/unit/type/exec_spec.rb

```diff
@@ -396,18 +396,28 @@ def test(command, valid)
         end
       end
 
-      context 'when timeout is exceeded on POSIX', :as_platform => :posix do
-        it 'sends a SIGTERM and raises a Puppet::Error' do
+      describe 'when timeout is exceeded' do
+        subject do
           ruby_path = Puppet::Util::Execution.ruby_path()
 
           ## Leaving this commented version in here because it fails on windows, due to what appears to be
           ##  an assumption about hash iteration order in lib/puppet/type.rb#hash2resource, where
           ##  resource[]= will overwrite the namevar with ":name" if the iteration is in the wrong order
           #sleep_exec = Puppet::Type.type(:exec).new(:name => 'exec_spec sleep command', :command => "#{ruby_path} -e 'sleep 0.02'", :timeout => '0.01')
```

Comment:
Two comments on comments :)
* maybe update the commented out line to match its fixed-up uncommented line (no `sleep_exec =`, same timeouts)
* add a comment about the timeout in the spec test needing to be long enough for ruby to fire up and spawn a child (since the timeout is a bit fidgety, this could crop up again six months from now, and that comment could be a timesaver for someone)